shared/health/src/test/java/io/pravega/shared/health/HealthServiceTests.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import io.pravega.test.common.AssertExtensions;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.rules.Timeout;
+
+import io.pravega.shared.health.TestHealthIndicators.SampleHealthyIndicator;
+
+@Slf4j
+// Hosts much of the same tests that would go into a HealthEndpointTests class, making HealthEndpointTests redundant.
+public class HealthServiceTests {
+
+    @Rule
+    public final Timeout timeout = new Timeout(600, TimeUnit.SECONDS);
+
+    HealthService service;
+
+    HealthServiceFactory factory;
+
+    public void start() throws IOException {
+        factory = new HealthServiceFactory();
+        service = factory.createHealthService(true);
+    }
+
+    public void stop() {
+        service.clear();
+        factory.close();
+        Assert.assertEquals("The HealthService should not maintain any references to HealthContributors.",
+                1,
+                service.components().size());
+        Assert.assertEquals("The ContributorRegistry should not maintain any references to HealthContributors",
+                1,
+                service.registry().contributors().size());
+    }
+
+    @Before
+    @SneakyThrows
+    public void before() {
+        start();
+    }
+
+    @After
+    public void after() {
+        stop();
+
+    }
+
+    @Test
+    public void testLifecycle() throws IOException {
+        // Perform a start-up and shutdown sequence (implicit start() with @Before).
+        stop();
+        // Verify that it is repeatable.
+        start();
+        stop();
+    }
+
+    @Test
+    public void testHealth() throws IOException {
+        SampleHealthyIndicator indicator = new SampleHealthyIndicator();
+        service.registry().register(indicator);
+        try {
+            Health health = service.endpoint().health(indicator.getName());
+            Assert.assertTrue("Status of the default/root component is expected to be 'UP'", health.getStatus() == Status.UP);
+
+            health = service.endpoint().health(true);
+            Assert.assertEquals("There should be exactly one child (SimpleIndicator)", health.getChildren().size(), 1);
+        } catch (ContributorNotFoundException e) {
```

Comment:
You should just leave this to bubble up. You'll get better messages in your tests along with  full stack trace.